iwyu_test_util.py

```diff
@@ -503,6 +512,47 @@ def _CheckPrerequisites(cc_file):
     raise unittest.SkipTest(', '.join(failed))
 
 
+def _ParseExpectedFailures(cc_file):
+  """ Parses expected failure conditions out of cc_file. """
+  conditions = []
+  with open(cc_file) as it:
+    for lineno, line in enumerate(it):
+      m = _IWYU_TEST_XFAIL_RE.match(line)
+      if not m:
+        continue
+
+      feature = m.group(1)
+      value = m.group(2)
+
+      if feature is None:
+        # Bare IWYU_XFAIL, condition is always True
+        condition = lambda: True
+        desc = 'always'
+
+      else:
+        feature_cond = _FEATURES.known.get(feature)
+        if not feature_cond:
+          raise SyntaxError('%s:%s IWYU_XFAIL: unsupported feature: %s' %
+                            (cc_file, lineno, feature))
+
+        condition = lambda arg=value: feature_cond(arg)
+        desc = '%s(%s)' % (feature, value)
+
+      conditions.append((condition, desc))
+    return conditions
+
+
+def IsTestExpectedToFail(cc_file):
+  """ Checks if a test is expected to fail. """
+  xfail_conditions = _ParseExpectedFailures(cc_file)
+
+  for condition, _ in xfail_conditions:
```

Comment:
You could consider `return any(condition() for condition, _ in xfail_conditions)`; then this would be a two-line function.

Reply:
But if it looks too opaque to you (I'm on the fence), please remove the empty lines in this functions